### PR TITLE
Adds Support to Visual Studio Code Dev Container

### PR DESCRIPTION
**Pull request by AlexCatarino**
#### Description
Adds support and documentation to VSCode Dev Container.

#### Motivation and Context
Help the Lean developer community.

#### How Has This Been Tested?
Running Lean in Dev Container with C# and Python backtests.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: .vscode/readme.md [hunk 1]
```diff
@@ -12,16 +14,47 @@ This document contains information regarding ways to use Visual Studio Code to w
...
+5. Open Development Container
+    - In VS Code, either:
+        - Select "Reopen in Container" from pop up box.
+
+            OR
+
+        - Ctrl+Shift+P (Command Palette) and select "Remote-Containers: Rebuild and Reopen in Container"
+
+You should now be in the development container, give VS Code a moment to prepare and you will be ready to go!
+If you would like to mount any additional local files to your container, checkout devcontainer.json "mounts" section for an example! Upon any mount changes you must rebuild the container using Command Palette as in step 5.
```
Comment (Martin-Molinero) [inline .vscode/readme.md:48]: > checkout devcontainer.json

Maybe this could be a link?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	},
+
+	// Add the IDs of extensions you want installed when the container is created.
+	"extensions": ["ms-dotnettools.csharp", "ms-python.python", "ms-python.vscode-pylance", "formulahendry.dotnet-test-explorer", "eamodio.gitlens", "yzhang.markdown-all-in-one"],
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	// "forwardPorts": [],
+
+	// Uncomment the next line to run commands after the container is created - for example installing curl.
+	"postCreateCommand": "dotnet nuget add source /Lean/LocalPackages",
```
Comment (Martin-Molinero) [inline .devcontainer/devcontainer.json:27]: Should we add `dotnet build` too?

File: .vscode/tasks.json [hunk 3]
```diff
@@ -5,6 +5,25 @@
...
+            "type": "shell",
+            "isBackground": true,
+            "windows": {
+                "command": "${workspaceFolder}/.vscode/autobuilder.bat",
+            },
+            "linux": {
+                "command": "${workspaceFolder}/.vscode/autobuilder.sh"
+            },
+            "osx": {
+                "command": "${workspaceFolder}/.vscode/autobuilder.sh"
```
Comment (Martin-Molinero) [inline .vscode/tasks.json:20]: Do we need these autobuilders? 🤔

File: .vscode/launch_research.sh [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    \"data-folder\": \"${DATA_DIR}\",
+    \"composer-dll-directory\": \"${BUILD_DIR}\",
+    \"algorithm-language\": \"Python\",
+    \"messaging-handler\": \"QuantConnect.Messaging.Messaging\",
+    \"job-queue-handler\": \"QuantConnect.Queues.JobQueue\",
+    \"api-handler\": \"QuantConnect.Api.Api\",
+    \"job-user-id\": \"0\",
+    \"api-access-token\": \"\",
+    }" > $CONFIG
+fi
```
Comment (Martin-Molinero) [inline .vscode/launch_research.sh:44]: The config.json will already be in the output folder of the launcher, same as research itself, we don't really need this?

It failed for me
```
/Lean/.vscode/launch_research.sh: line 2: syntax error near unexpected token `$'{\r''
'Lean/.vscode/launch_research.sh: line 2: `absolute_path() {

 *  The terminal process "/usr/bin/bash '-c', '/Lean/.vscode/launch_research.sh /Lean/Launcher/bin/Debug /Lean/Data'" failed to launch (exit code: 2). 
```

File: .vscode/tasks.json [hunk 5]
```diff
@@ -50,6 +69,18 @@
...
+            "label": "start research",
+            "type": "shell",
+            "group": "build",
+            "isBackground": true,
+            "command" : "${workspaceFolder}/.vscode/launch_research.sh",
+            "args" : [
+                "${workspaceFolder}/Launcher/bin/Debug",
+                "${workspaceFolder}/Data"
+            ],
+            "problemMatcher": "$msCompile"
```
Comment (Martin-Molinero) [inline .vscode/tasks.json:83]: Which should probably add a dependency on the build task for this one? will it open the jupyter notebook viewer automatically?

File: .vscode/launch_research.sh [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+# Use our startup script for jupyter, needed to set runtime properly
+mkdir -p /root/.ipython/profile_default/startup/ && \
+ln -s /Lean/Launcher/bin/Debug/start.py /root/.ipython/profile_default/startup/start.py
```
Comment (Martin-Molinero) [inline .vscode/launch_research.sh:48]: Isn't this already in the base research image?

File: .vscode/tasks.json (hunk 5 above)
Comment (AlexCatarino) [reply .vscode/tasks.json:64]: After we run this command, a pop-up window opens and we can open the notebook on a web browser.

**Review by Martin-Molinero — APPROVED** [review]
Sweet!

**Pull request merged**